### PR TITLE
Updated win_disk_facts module

### PR DESCRIPTION
**Pull request by marqelme**
##### SUMMARY
- Added trimmed option to module win_disk_facts
- Removed required administrator privileges

The trimmed option can be used to reduce the output of the module.
Then only the disk fact details will be displayed but no detailed information about the partitions and volumes of the disk.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
win_disk_facts

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Nov 20 2017, 18:23:56) [GCC 5.4.0 20160609]

```

### CODE REVIEW

**Comment by dagwieers** [comment]
This PR is undoing a lot of things we recently added for the v2.5 release.

I am also not convinced we want an option "trimmed" here. This requires a discussion, both for the reasons as well as what would be an acceptable interface.

File: lib/ansible/modules/windows/win_disk_facts.py [hunk 1]
```diff
@@ -8,47 +8,41 @@
...
-    disksize_gib: '{{ (disk.size/1024|pow(3))|round|int }} GiB'  # returns "224 GiB" (value in GiB)
-
-    # Show disk size in Gigabytes
-    disksize_gb_human: '{{ disk.size|filesizeformat }}'        # returns "240.1 GB" (human readable)
-    disksize_gb: '{{ (disk.size/1000|pow(3))|round|int }} GB'  # returns "240 GB" (value in GB)
-
-- name: Output second disk serial number
+- name: get disk facts
+  win_disk_facts:
+- name: output second disk serial number
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_disk_facts.py]: This is undoing the example.

File: lib/ansible/modules/windows/win_disk_facts.py [hunk 2]
```diff
@@ -59,7 +53,12 @@
     returned: always
     type: complex
     contains:
-        ansible_disks:
+        total_disks:
+            description: Count of found disks on the target.
+            returned: if disks were found
+            type: int
+            sample: 3
+        disks:
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_disk_facts.py]: This is undoing the explicit ansible_disk key.

File: lib/ansible/modules/windows/win_disk_facts.py [hunk 3]
```diff
@@ -70,10 +69,10 @@
                     type: int
                     sample: 0
                 size:
-                    description: Size in bytes of the particular disk.
+                    description: Size in Gibibyte of the particular disk.
                     returned: always
-                    type: int
-                    sample: 227727638528
+                    type: string
+                    sample: "100GiB"
```
Comment (dagwieers) [inline lib/ansible/modules/windows/win_disk_facts.py]: We wanted raw values on purpose.

**Review by dagwieers — CHANGES_REQUESTED** [review]

**Comment by marqelme** [comment]
@dagwieers 
Yes, this was a mistake, I have corrected it.

Sure, please discuss this, just thought this option would help because the module creates a lot of output if you have a lot disks, partitions and volumes.

**Comment by varunchopra** [comment]
Is there a reason we're not treating volumes and disks differently? I thought we arrived at a conclusion over at https://github.com/ansible/ansible/pull/27634 that it's better to create different modules.

IMHO this is way too bloated and all I'm all against the disk_facts module. :/

**Comment by dagwieers** [comment]
Beware that the deadline for getting new features/modules accepted in [Ansible v2.6 is nearing](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_6.rst#release-schedule), it is set to either 2018-05-17 or 2018-05-25. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to [the Windows Working Group meeting agenda](https://github.com/ansible/community/issues/294) to get it resolved.

**Comment by dagwieers** [comment]
Beware that the deadline for getting new features/modules accepted in [Ansible v2.7 is nearing](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_7.rst#release-schedule), it is set to either 2018-08-23 or 2018-08-30. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to the [Windows Working Group meeting agenda](https://github.com/ansible/community/issues/294) to get it resolved.

**Comment by taschaal** [comment]
I don't see it mentioned anywhere, but this module doesn't appear to work with windows dynamic disks.  Am I correct in that?

**Comment by marqelme** [comment]
Currently I am not sure if I have tried this.
Have you tried to gather the dynamic disks with the module?
Otherwise I will try it by tomorrow.

Outlook für iOS<https://aka.ms/o0ukef> beziehen

________________________________
Von: taschaal <notifications@github.com>
Gesendet: Montag, November 5, 2018 6:52 PM
An: ansible/ansible
Cc: Marc Tschapek; Author
Betreff: Re: [ansible/ansible] Updated win_disk_facts module (#37798)


I don't see it mentioned anywhere, but this module doesn't appear to work with windows dynamic disks. Am I correct in that?

**Comment by it-praktyk** [comment]
@marqelme, could you consider rebasing your fork? I tried to do that by myself and make PR to your repository but - due to a number of commits - it's impossible (a web browser friezes).

I'm interested to help finalize that update.

**Comment by it-praktyk** [comment]
@taschaal, @marqelme the issue it's not about dynamic but about the bug #51032.

**Comment by it-praktyk** [comment]
@taschaal, @marqelme I was wrong, the module doesn't handle dynamic disks - details included in the #51032.

**Comment by it-praktyk** [comment]
I've analyzed what is the real subject of the proposed change and the resulted JSON files are available [here](https://gist.github.com/it-praktyk/ab42cda9656425d7f5251a98c582bebf) as the gist.

IMHO, that change should be abandoned.

**Comment by chopraaa** [comment]
I think instead of `trimmed`, we can specify what kind of facts we want, such as volumes or partitions.

```yaml
win_disk_facts:
    type: volume
```

Should return facts relevant only to volumes instead of flooding a single variable with stuff we don't really need.

**Comment by jborean93** [comment]
Due to the age of the PR and the conflicts with the win_disk_facts module I am going to close this PR. If you still wish to include this change in some form or another please open a new PR against the latest branch of Ansible.

Really sorry for the delay in getting to these PRs and I am hoping we can be a lot more responsive in the future.

**Pull request closed without merging**